label_studio_sdk/project.py

```diff
@@ -520,16 +521,26 @@ def get_tasks(
             Task list with task data, annotations, predictions and other fields from the Data Manager
 
         """
-        data = self.get_paginated_tasks(
-            filters=filters,
-            ordering=ordering,
-            view_id=view_id,
-            selected_ids=selected_ids,
-            only_ids=only_ids,
-            page=1,
-            page_size=-1
-        )
-        return data['tasks']
+
+        page = 1
+        result = []
+        while True:
+            try:
+                data = self.get_paginated_tasks(
+                    filters=filters,
+                    ordering=ordering,
+                    view_id=view_id,
+                    selected_ids=selected_ids,
+                    only_ids=only_ids,
+                    page=page,
+                    page_size=100
+                )
+                result += data['tasks']
+                page += 1
+            # we'll get 404 from API on empty page
+            except LabelStudioException:
```

Comment:
we need logging here, and a more broad exception handler should prevent infinite loop

Reply:
> we need logging here

It's not an error, we expect, that after the last page we'll get 404, what type of logging it should be? 

> more broad exception handler should prevent infinite loop

I can't see ability for infinite loop there, for LabelStudioException we have a `break` statement and any other exception will throw up and interrupt the loop.

Reply:
Added logging.